core/src/main/java/com/linecorp/armeria/server/RouteCache.java

```diff
@@ -142,7 +148,7 @@ private RouteCache() {}
             }
 
             final Routed<V> result = delegate.find(routingCtx);
-            if (result.isPresent()) {
+            if (result.isPresent() && !noCacheRoutes.contains(result.route())) {
```

Comment:
I'm somewhat concerned about the performance implication of this check, because it will involve at least the comparison of 3 collections and 1 `PathMapping`. Would it still work if we use a set with identity comparison?

Reply:
Agreed. The no cache routes may be matched by identity comparison. So I think it would work. Let me change the set to Set<Integer> which store Route's hash code. Does it make sense?

Reply:
I'm not sure I understand `noCacheRoutes` correctly. But don't we need to add 
`!noCacheRoutes.contains(result.route()))` predicate to `findAll(RoutingContext)` method?

Reply:
@ikhoon I think you're right. I didn't think about that because there was no `findAll` method when I started this PR. :-( Let me figure out how I handle it. Thanks!

Reply:
@ikhoon Fixed. :-)